modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/distruptor/inbound/ConcurrentBatchEventHandler.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.andes.kernel.distruptor.inbound;
+
+import com.lmax.disruptor.*;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This is a turn based concurrent event batching handler. This should be used with a Default Batch processor
+ * The processor will batch event until end of current available sequence numbers is visited send the events
+ * to this concurrent event batch handler to do custom event batching
+ * 
+ * NOTE: Writing a custom batch processor is avoided since it related to implementing disruptor internals related
+ * logic which might lead to difficulty in upgrading disruptor versions and getting bug fixes on batch processors 
+ */
+public class ConcurrentBatchEventHandler implements EventHandler<InboundEvent> {
+
+    private static Log log = LogFactory.getLog(ConcurrentBatchEventHandler.class);
+
+
+    private final BatchEventHandler eventHandler;
+    private final long turn;
+    private final int groupCount;
+    private int batchSize;
```

Comment:
Shall we explain these variables ?!!! so others can understand the magic you guys brought in!